packages/yourloops/test/integration/assert/caregivers-list.assert.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { screen, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { mockCaregiverUser } from '../mock/direct-share.api.mock'
+import NotificationApi from '../../../lib/notifications/notification.api'
+import { NotificationType } from '../../../lib/notifications/models/enums/notification-type.enum'
+import { type Notification } from '../../../lib/notifications/models/notification.model'
+import DirectShareApi, { PATIENT_CANNOT_BE_ADDED_AS_CAREGIVER_ERROR_MESSAGE } from '../../../lib/share/direct-share.api'
+import { UserInviteStatus } from '../../../lib/team/models/enums/user-invite-status.enum'
+
+export const checkCaregiversListLayout = async () => {
+  expect(await screen.findByTestId('patient-caregivers-list')).toBeVisible()
+
+  const addCaregiverButton = await screen.findByTestId('add-caregiver-button')
+  expect(addCaregiverButton).toBeVisible()
+  expect(addCaregiverButton).toBeEnabled()
+}
+
+export const checkAddRemoveCaregiver = async () => {
+  const caregiversTable = screen.getByTestId('patient-caregivers-list')
+  expect(caregiversTable).toHaveTextContent('Last nameFirst nameEmail')
+
+  const caregiverRow = within(caregiversTable).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}`)
+  expect(caregiverRow).toBeVisible()
+  expect(caregiverRow).toHaveTextContent('UserCaregivercaregiver@mail.com')
+  expect(within(caregiverRow).getByTestId(`patient-caregivers-table-row-${mockCaregiverUser.username}-button-remove`)).toBeVisible()
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏Checking the content of the caregiver list could have been done in `checkCaregiversListLayout` as this is not related to caregiver removal (it could be justified that it is in the case that the data was not mocked)